sql/2020/16_Ecommerce/android_ios_app_links_ecomm_sites.sql

```diff
@@ -0,0 +1,26 @@
+#standardSQL
+# 13_21: Native app links association for mobile ecommerce sites. 
+# Querying against mobile dataset as apps are relevant in context of mobile. This query uses custom metric 'ecommerce' - https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/ecommerce.js
+SELECT
+  COUNTIF(android_app_links) AS android_app_links,
+  COUNTIF(ios_universal_links) AS ios_universal_links,
+  COUNT(0) AS total,
+  COUNTIF(android_app_links) / COUNT(0) AS pct_android_app_links,
+  COUNTIF(ios_universal_links) / COUNT(0) AS pct_ios_universal_links
+FROM (
+  SELECT DISTINCT
+    url
+  FROM
+    `httparchive.technologies.2020_08_01_mobile`
+  WHERE
+    category = 'Ecommerce')
+JOIN (
+  SELECT
+    url,
+    JSON_EXTRACT(JSON_EXTRACT_SCALAR(payload, '$._ecommerce'), '$.AndroidAppLinks') = '1' AS android_app_links,
+    JSON_EXTRACT(JSON_EXTRACT_SCALAR(payload, '$._ecommerce'), '$.iOSUniveralLinks') = '1' AS ios_universal_links
+  FROM
+    `httparchive.pages.2020_08_01_mobile`)
+USING
+  (url)
+
```

Comment:
As discussed on chat let's look at mobile and desktop just in case it throws up any interesting stats (e.g. desktop with higher % of links - though it doesn't in this case). Can then only report on mobile in chapter.

```suggestion
# 13_21: Native app links association for ecommerce sites. 
# This query uses custom metric 'ecommerce' - https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/ecommerce.js
SELECT
  client,
  COUNTIF(android_app_links) AS android_app_links,
  COUNTIF(ios_universal_links) AS ios_universal_links,
  COUNT(0) AS total,
  COUNTIF(android_app_links) / COUNT(0) AS pct_android_app_links,
  COUNTIF(ios_universal_links) / COUNT(0) AS pct_ios_universal_links
FROM (
  SELECT DISTINCT
    _TABLE_SUFFIX as client,
    url
  FROM
    `httparchive.technologies.2020_08_01_*`
  WHERE
    category = 'Ecommerce')
JOIN (
  SELECT
    _TABLE_SUFFIX as client,
    url,
    JSON_EXTRACT(JSON_EXTRACT_SCALAR(payload, '$._ecommerce'), '$.AndroidAppLinks') = '1' AS android_app_links,
    JSON_EXTRACT(JSON_EXTRACT_SCALAR(payload, '$._ecommerce'), '$.iOSUniveralLinks') = '1' AS ios_universal_links
  FROM
    `httparchive.pages.2020_08_01_*`)
USING
  (client, url)
GROUP BY
  client
ORDER BY
  client

```

Reply:
Accepting this suggestion and updated the results from modified query in Excel sheet.